### PR TITLE
tree: Fix invalid unlocked_by reference panic 

### PR DESCRIPTION
**Pull request by tleen**
See #102

### CODE REVIEW

**Comment by rpottsoh** [comment]
Thanks @tleen for making the necessary changes.

File: cmd/tree.go [hunk 1]
```diff
@@ -230,8 +230,17 @@ func treeTrack(configFilepath string) error {
 			continue
 		}
 
+		parent, unlockExists := slugToExercise[e.UnlockedBy]
```
Comment (nywilken) [inline cmd/tree.go:233]: Do we need to run this type of exists check in other areas of the code where we expect to get a reference to an exercise?

**Review by ferhatelmas — CHANGES_REQUESTED** [review]
Seems test is missing `gofmt`. Otherwise good.

**Review by nywilken — APPROVED** [review]
@tleen this looks good to me. I am a little concerned that the warning may get missed, especially for tracks with a large set of exercises.

I left a question about your change. But this is otherwise good to merge, when you are ready.

File: cmd/tree_example_test.go [hunk 2]
```diff
@@ -84,3 +84,27 @@ func ExampleTreeWarnings() {
...
+	// ----
+	// ├─ one
+	// │  └─ five
+	// │
+	// └─ two
+	//
+	// bonus
+	// -----
+	// seven
+}
```
Comment (tleen) [inline cmd/tree_example_test.go]: @ferhatelmas is this the line that makes you think it did not pass through `gofmt`?

Comment (tleen) [reply cmd/tree_example_test.go]: @ferhatelmas good eye on that, there are a few places it was wrong. Dunno why it didn't run on save as usual.

File: cmd/tree.go (hunk 1 above)
Comment (tleen) [reply cmd/tree.go:233]: I don't believe so, not in `tree` anyway. The rest are referenced via the main slug which is the authoritative map key for an exercise. 

Unsure about other commands. I think once `lint` checks for this stuff it is ok for other commands to assume that config won't have bad slugs.

Comment (tleen) [reply cmd/tree.go:233]: @nywilken the warning could be missed, but not sure how to make it stand out more?

File: cmd/tree.go [hunk 3]
```diff
@@ -230,8 +230,17 @@ func treeTrack(configFilepath string) error {
...
+		// An unlocked_by slug that does not exist and is referenced can crash
+		// the program, see #102. If an non-existant slug exists issue warning.
+		if !unlockExists{
+			printConfigurationWarning(
+				fmt.Sprintf("Exercise \"%s\" has an invalid unlocked_by slug: \"%s\"", e.Slug, e.UnlockedBy));
```
Comment (ferhatelmas) [inline cmd/tree.go]: `s/%s/%q/` seems better for formatting.

File: cmd/tree_example_test.go [hunk 4]
```diff
@@ -84,3 +84,27 @@ func ExampleTreeWarnings() {
 	// one
 	// two
 }
+
+func ExampleInvalidUnlockedBy(){
```
Comment (ferhatelmas) [inline cmd/tree_example_test.go]: Sorry about nitpick. There should be a space after parens.

Comment (tleen) [reply cmd/tree_example_test.go]: No you're right, something has gone wonky with my `gofmt`. It did everything but that.

File: cmd/tree.go (hunk 3 above)
Comment (tleen) [reply cmd/tree.go]: :100: better. Great call.

File: cmd/tree.go [hunk 5]
```diff
@@ -230,8 +230,17 @@ func treeTrack(configFilepath string) error {
 			continue
 		}
 
+		parent, unlockExists := slugToExercise[e.UnlockedBy]
+		// An unlocked_by slug that does not exist and is referenced can crash
+		// the program, see #102. If an non-existant slug exists issue warning.
```
Comment (ferhatelmas) [inline cmd/tree.go]: Also, `a non-existent slug` ? https://en.wiktionary.org/wiki/non-existant

Comment (tleen) [reply cmd/tree.go]: heh, a non-existant word.

File: cmd/tree.go [hunk 6]
```diff
@@ -232,7 +232,7 @@ func treeTrack(configFilepath string) error {
 
 		parent, unlockExists := slugToExercise[e.UnlockedBy]
 		// An unlocked_by slug that does not exist and is referenced can crash
-		// the program, see #102. If an non-existant slug exists issue warning.
+		// the program, see #102. If an non-existent slug exists issue warning.
```
Comment (ferhatelmas) [inline cmd/tree.go]: `a non-...` :wink:

**Review by ferhatelmas — APPROVED** [review]

**Pull request merged**